spring-cloud-gcp-pubsub/src/test/java/org/springframework/cloud/gcp/pubsub/core/subscriber/PubSubSubscriberTemplateTests.java

```diff
@@ -114,29 +185,239 @@ public void setUp() {
 	}
 
 	@Test
-	public void testSubscribeAndConvert() {
+	public void testSubscribeAndAck() {
```

Comment:
A bunch of tests below look almost identical with a lot of code duplication. Can we reduce the code duplication? We can  add utility methods and/or introduce a reusable test method that accepts a strategy for behavior that varies among the tests.
We can also just leave this to a future refactoring.

Reply:
Agreed.  If the tests are valid, then I would like to open an issue to refactor them and defer so we can make sure to get this code in sooner than later.